redis/src/aio/pubsub.rs

```diff
@@ -297,28 +297,44 @@ impl PubSubSink {
     pub async fn subscribe(&mut self, channel_name: impl ToRedisArgs) -> RedisResult<()> {
         let mut cmd = cmd("SUBSCRIBE");
         cmd.arg(channel_name);
-        self.send_recv(cmd.get_packed_command()).await
+        self.send_recv(cmd.get_packed_command()).await.map(|_| ())
     }
 
     /// Unsubscribes from channel.
     pub async fn unsubscribe(&mut self, channel_name: impl ToRedisArgs) -> RedisResult<()> {
         let mut cmd = cmd("UNSUBSCRIBE");
         cmd.arg(channel_name);
-        self.send_recv(cmd.get_packed_command()).await
+        self.send_recv(cmd.get_packed_command()).await.map(|_| ())
     }
 
     /// Subscribes to a new channel with pattern.
     pub async fn psubscribe(&mut self, channel_pattern: impl ToRedisArgs) -> RedisResult<()> {
         let mut cmd = cmd("PSUBSCRIBE");
         cmd.arg(channel_pattern);
-        self.send_recv(cmd.get_packed_command()).await
+        self.send_recv(cmd.get_packed_command()).await.map(|_| ())
     }
 
     /// Unsubscribes from channel pattern.
     pub async fn punsubscribe(&mut self, channel_pattern: impl ToRedisArgs) -> RedisResult<()> {
         let mut cmd = cmd("PUNSUBSCRIBE");
         cmd.arg(channel_pattern);
-        self.send_recv(cmd.get_packed_command()).await
+        self.send_recv(cmd.get_packed_command()).await.map(|_| ())
+    }
+
+    /// Sends a ping with a message to the server
+    pub async fn ping_message<T: FromRedisValue>(
+        &mut self,
+        message: impl ToRedisArgs,
+    ) -> RedisResult<T> {
+        let mut cmd = cmd("PING");
```

Comment:
Is there a reason to use two lines with `mut` and not `cmd("PING").arg(message);`?
If the goal is readability, the extra verbosity needed is taking `cmd.get_packed_command()).await?` out, and giving the result as a parameter to `from_owned_redis_value`. 
Current style complicating the code, IDK what you expect from `cmd.get_packed_command()).await?`, the parameter itself is too long and not fun to look at.
Giving it a name will enhance the ability to understand what is the value you want to pass.
There is a hidden assumption that the next developer working on this code already knows all the functions and their results, while that's not the case if I would want to add some code to this part.

Reply:
I went with how the previous commands are written, no big reason.

`let cmd = cmd("PING").arg(message);` doesn't compile, because it keeps a reference to a temp value.
I can do `let cmd = cmd("PING").arg(message).get_packed_command();`